File: libcrocofixpython_tests/field.py
import unittest
from crocofix import *

class TestField(unittest.TestCase):

    def test_timestamp_format_enum(self):
        TimestampFormat.__members__ == {
            'SECONDS': TimestampFormat.SECONDS,
            'MILLISECONDS': TimestampFormat.MILLISECONDS
        }
    
    def test_timestamp_format_seconds(self):
        self.assertEqual(len(timestamp_string(TimestampFormat.SECONDS)), 17)
 
    def test_timestamp_format_milliseconds(self):
        self.assertEqual(len(timestamp_string(TimestampFormat.MILLISECONDS)), 21)
 
    def test_init(self):
        self.assertEqual(Field(0, True).value, "Y")
        self.assertEqual(Field(0, False).value, "N")
        self.assertEqual(Field(0, "value").value, "value")
        self.assertEqual(Field(0, 1.2345).value, "1.234500")
        self.assertEqual(Field(0, -1.2345).value, "-1.234500")
        self.assertEqual(Field(0, 567800).value, "567800")
        self.assertEqual(Field(0, -567800).value, "-567800")
        
if __name__ == "__main__":
    unittest.main()

File: libcrocofixpython/field.cpp
#include "field.hpp"

import crocofix;

void init_field(py::module_& module)
{
    py::enum_<crocofix::timestamp_format>(module, "_TimestampFormat")
        .value("SECONDS", crocofix::timestamp_format::seconds)
        .value("MILLISECONDS", crocofix::timestamp_format::milliseconds)
    ;

    module.def("_timestamp_string", &crocofix::timestamp_string);

    py::class_<crocofix::field>(module, "_Field")
        .def(py::init<int, std::string>())
        .def(py::init<int, bool>())
        .def(py::init<int, double>())
        .def(py::init<int, int64_t>())
        .def_property_readonly("tag", &crocofix::field::tag)
        .def_property_readonly("value", &crocofix::field::value)
    ;

}